infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -71,7 +71,7 @@ private long getLogSizeLimit() {
 
         // Derived size in bytes that normal writes to the log unit are capped at.
         // This is derived as a percentage of the log's filesystem capacity.
-        return (long) (fileSystemCapacity * config.limitPercentage / 100);
+        return (long) ((fileSystemCapacity - config.orthogonalData * 1000) * config.limitPercentage / 100);
```

Comment:
That calculation is incorrect.

` long meaningfulName1 = fileSystemCapacity - config.orthogonalData * 1000;
`

It will overflow if config.orthogonalData = Integer.MAX_VALUE . You need to cast to a long: 

` long meaningfulName1 = fileSystemCapacity - config.orthogonalData * 1000L;
`